Changelog entry for v0.11.0 (#3022)
GenericMappingTools/pygmt: doc/changes.md

# Changelog

## Release v0.11.0 (2024/02/01)

[![Digital Object Identifier for PyGMT v0.11.0](https://zenodo.org/badge/DOI/10.5281/zenodo.10578540.svg)](https://doi.org/10.5281/zenodo.10578540)

> ### Highlights
> 
> * Tutorial for table inputs ([#2722](https://github.com/GenericMappingTools/pygmt/pull/2722)) and gallery example for choropleth map ([#2796](https://github.com/GenericMappingTools/pygmt/pull/2796))

Reviewer: ```suggestion
### Highlights

* 🎉 **Eleventh minor release of PyGMT** 🎉
* Tutorial for table inputs ([#2722](https://github.com/GenericMappingTools/pygmt/pull/2722)) and gallery example for choropleth map ([#2796](https://github.com/GenericMappingTools/pygmt/pull/2796))
```

> * Easy access to remote datasets of planetary reliefs (Moon, Venus, Mars, Mercury, Pluto) ([#2674](https://github.com/GenericMappingTools/pygmt/pull/2674), [#2906](https://github.com/GenericMappingTools/pygmt/pull/2906), [#2847](https://github.com/GenericMappingTools/pygmt/pull/2847))

Author: ```suggestion
* Easy access to remote datasets of planetary reliefs (Moon, Venus, Mars, Mercury, Pluto) ([#2674](https://github.com/GenericMappingTools/pygmt/pull/2674), [#2906](https://github.com/GenericMappingTools/pygmt/pull/2906), [#2847](https://github.com/GenericMappingTools/pygmt/pull/2847), [#3027](https://github.com/GenericMappingTools/pygmt/pull/3027), [#3028](https://github.com/GenericMappingTools/pygmt/pull/3028))
```

Reviewer: Did you want the Mercury and Pluto datasets in v0.11.0? Or push it to the next release?

Author: Prefer to have them in v0.11.0 so that all new platenary datasets are available via PyGMT.

> * Easy access to planetary relief datasets (Moon, Venus, Mars, Mercury, Pluto) ([#2674](https://github.com/GenericMappingTools/pygmt/pull/2674), [#2906](https://github.com/GenericMappingTools/pygmt/pull/2906), [#2847](https://github.com/GenericMappingTools/pygmt/pull/2847), [#3027](https://github.com/GenericMappingTools/pygmt/pull/3027), [#3028](https://github.com/GenericMappingTools/pygmt/pull/3028))

Reviewer: Sort by proximity to the Sun.

```suggestion
* Easy access to planetary relief datasets (Mercury, Venus, Moon, Mars, Pluto) ([#3028](https://github.com/GenericMappingTools/pygmt/pull/3028),  [#2906](https://github.com/GenericMappingTools/pygmt/pull/2906), [#2674](https://github.com/GenericMappingTools/pygmt/pull/2674), [#2847](https://github.com/GenericMappingTools/pygmt/pull/2847), [#3027](https://github.com/GenericMappingTools/pygmt/pull/3027), )
```

* Figure.timestamp: Deprecate parameter 'justification' to 'justify' (remove in v0.13.0) ([#3002](https://github.com/GenericMappingTools/pygmt/pull/3002))
* Figure.grdimage: Remove the unsupported 'img_out'/'A' parameter ([#2907](https://github.com/GenericMappingTools/pygmt/pull/2907))

### Bug Fixes

* pygmt.which: Fix the bug when passing multiple files ([#2726](https://github.com/GenericMappingTools/pygmt/pull/2726))
* pygmt.filter1d: Fix the bug that the first line is read as headers ([#2780](https://github.com/GenericMappingTools/pygmt/pull/2780))
* clib: Fix the bug when passing multiple columns of strings with variable lengths to the GMT C API ([#2719](https://github.com/GenericMappingTools/pygmt/pull/2719))
* Let kwargs_to_strings work with default values and positional arguments ([#2826](https://github.com/GenericMappingTools/pygmt/pull/2826))
* Figure.meca: Fix typo pricipal_axis -> principal_axis ([#2940](https://github.com/GenericMappingTools/pygmt/pull/2940))

### Documentation

* Use consistent description for the 'outgrid' parameter ([#2874](https://github.com/GenericMappingTools/pygmt/pull/2874))

> * Switch away from Stamen basemaps ([#2717](https://github.com/GenericMappingTools/pygmt/pull/2717))

Reviewer 2: Maybe we should move this to the Maintenance section?

* Recommend Miniforge instead of Mambaforge ([#2833](https://github.com/GenericMappingTools/pygmt/pull/2833))
* Improve the onboarding access checklist for contributors, maintainers and administrators ([#2656](https://github.com/GenericMappingTools/pygmt/pull/2656))
* Gallery example "Legend": Update regarding input data and multi-column legends ([#2762](https://github.com/GenericMappingTools/pygmt/pull/2762))

> * Figure.shift_origin: Improve docstrings, add inline examples and add type hints ([#2879](https://github.com/GenericMappingTools/pygmt/pull/2879))

Reviewer 2: Maybe we should move this to the Maintenance section to the part with the other type hints PRs?

> * Use consistent description for the 'outgrid' parameter ([#2874](https://github.com/GenericMappingTools/pygmt/pull/2874))
> * Switch away from Stamen basemaps ([#2717](https://github.com/GenericMappingTools/pygmt/pull/2717))
> * Recommend Miniforge instead of Mambaforge ([#2833](https://github.com/GenericMappingTools/pygmt/pull/2833))
> * Improve the onboarding access checklist for contributors, maintainers and administrators ([#2656](https://github.com/GenericMappingTools/pygmt/pull/2656))
> * Gallery example "Legend": Update regarding input data and multi-column legends ([#2762](https://github.com/GenericMappingTools/pygmt/pull/2762))
> * Figure.shift_origin: Improve docstrings, add inline examples and add type hints ([#2879](https://github.com/GenericMappingTools/pygmt/pull/2879))
> * Figure.plot: Update docstring for "connection" parameter to GMT 6.5 ([#2994](https://github.com/GenericMappingTools/pygmt/pull/2994))
> * Add the Japanese "PyGMT-HOWTO" tutorial to "External Resources" ([#2743](https://github.com/GenericMappingTools/pygmt/pull/2743))
> * Add gallery example showing how to adjust line segment ends (caps and joints) ([#3015](https://github.com/GenericMappingTools/pygmt/pull/3015))
> * Add gallery example for plotting connection lines ("connection" parameter of Figure.plot) ([#2999](https://github.com/GenericMappingTools/pygmt/pull/2999))
> * Add gallery example "Scale bar" ([#2822](https://github.com/GenericMappingTools/pygmt/pull/2822))

Reviewer 2: Remove "Switch away from Stamen basemaps" and "Figure.shift_origin: Improve docstrings, add inline examples and add type hints" and do some sorting. Feel free to make further adjustments to the sorting.

```suggestion
* Add gallery example "Scale bar" ([#2822](https://github.com/GenericMappingTools/pygmt/pull/2822))
* Add gallery example for plotting connection lines ("connection" parameter of Figure.plot) ([#2999](https://github.com/GenericMappingTools/pygmt/pull/2999))
* Add gallery example showing how to adjust line segment ends (caps and joints) ([#3015](https://github.com/GenericMappingTools/pygmt/pull/3015))
* Gallery example "Legend": Update regarding input data and multi-column legends ([#2762](https://github.com/GenericMappingTools/pygmt/pull/2762))
* Add the Japanese "PyGMT-HOWTO" tutorial to "External Resources" ([#2743](https://github.com/GenericMappingTools/pygmt/pull/2743))
* Figure.plot: Update docstring for "connection" parameter to GMT 6.5 ([#2994](https://github.com/GenericMappingTools/pygmt/pull/2994))
* Use consistent description for the "outgrid" parameter ([#2874](https://github.com/GenericMappingTools/pygmt/pull/2874))
* Improve the onboarding access checklist for contributors, maintainers and administrators ([#2656](https://github.com/GenericMappingTools/pygmt/pull/2656))
* Recommend Miniforge instead of Mambaforge ([#2833](https://github.com/GenericMappingTools/pygmt/pull/2833))
```

> ### Maintenance
> 
> * pyarrow: Check compatibility of pyarrow-backed pandas objects with numeric dtypes ([#2774](https://github.com/GenericMappingTools/pygmt/pull/2774))
> * doc: Convert the installation guides source code from ReST to Markdown ([#2992](https://github.com/GenericMappingTools/pygmt/pull/2992))
> * doc: Convert overview source code from ReST to Markdown ([#2953](https://github.com/GenericMappingTools/pygmt/pull/2953))
> * doc: Move compatibility table from README to separate file ([#2862](https://github.com/GenericMappingTools/pygmt/pull/2862))
> * Use ruff to lint and format codes, and remove flakeheaven/isort/black/blackdoc ([#2741](https://github.com/GenericMappingTools/pygmt/issues/2741))
> * Use codespell to check common misspellings ([#2673](https://github.com/GenericMappingTools/pygmt/pull/2673))
> * Use "# %%" as code block separators in examples ([#2662](https://github.com/GenericMappingTools/pygmt/pull/2662))
> * Update output shape and mean values from some x2sys_cross tests ([#2986](https://github.com/GenericMappingTools/pygmt/pull/2986))
> * TYP: Improve the doc style for type hints ([#2813](https://github.com/GenericMappingTools/pygmt/pull/2813))
> * TYP: Add type hints support for pygmt.datasets.load_sample_data ([#2859](https://github.com/GenericMappingTools/pygmt/pull/2859))
> * TYP: Add type hints for the 'registration' parameter in pygmt.datasets.load_* functions ([#2867](https://github.com/GenericMappingTools/pygmt/pull/2867))
> * TYP: Add type hints for 'data_source' in load_earth_relief and load_earth_magnetic_anomaly functions ([#2849](https://github.com/GenericMappingTools/pygmt/pull/2849))
> * Setup Continuous Benchmarking workflow with pytest-codspeed ([#2908](https://github.com/GenericMappingTools/pygmt/pull/2908))
> * Release Drafter: Automatically replace GitHub handles with names and links ([#2777](https://github.com/GenericMappingTools/pygmt/pull/2777))
> * Refactor the internal _load_remote_dataset function to simplify datasets' definitions ([#2917](https://github.com/GenericMappingTools/pygmt/pull/2917))
> * NEP29: Test PyGMT on NumPy 1.26 ([#2692](https://github.com/GenericMappingTools/pygmt/pull/2692))
> * Move variable `__gmt_version__` to pygmt.clib to avoid cyclic-import errors ([#2713](https://github.com/GenericMappingTools/pygmt/pull/2713))
> * Improve the way to import optional modules ([#2809](https://github.com/GenericMappingTools/pygmt/pull/2809))
> * Fix fixture_xr_image to open earth_day_01d_p directly with rioxarray ([#2963](https://github.com/GenericMappingTools/pygmt/pull/2963))
> * Figure.timestamp: Add type hints to parameters ([#2890](https://github.com/GenericMappingTools/pygmt/pull/2890))
> * Figure.solar: Add type hints to the 'terminator' parameter and simplify codes ([#2881](https://github.com/GenericMappingTools/pygmt/pull/2881))
> * Exclude CODE_OF_CONDUCT.md, AUTHORSHIP.md and pygmt/tests directory from distributions ([#2957](https://github.com/GenericMappingTools/pygmt/pull/2957))
> * Enable the PDF format documentation in the ReadTheDocs site ([#2876](https://github.com/GenericMappingTools/pygmt/pull/2876))
> * Create skip_if_no helper function to skip tests when missing a package ([#2883](https://github.com/GenericMappingTools/pygmt/pull/2883))
> * CI: Trigger the cache_data workflow in PRs if cache files are added/deleted/updated ([#2939](https://github.com/GenericMappingTools/pygmt/pull/2939))
> * CI: Test GMT dev version on Windows by building from source ([#2773](https://github.com/GenericMappingTools/pygmt/pull/2773))
> * CI: Set cache-downloads to false to speedup the "Setup Micromamba" step ([#2946](https://github.com/GenericMappingTools/pygmt/pull/2946))
> * CI: Run certain GitHub Actions workflows on official repo only ([#2951](https://github.com/GenericMappingTools/pygmt/pull/2951))
> * CI: Run benchmarks if PR is labeled with "run/benchmark" ([#2958](https://github.com/GenericMappingTools/pygmt/pull/2958))
> * CI: Run "GMT Dev Tests" if PR is labeled with 'run/test-gmt-dev' ([#2960](https://github.com/GenericMappingTools/pygmt/pull/2960))
> * CI: Require at least one code block separator for example files in the Style Checks workflow ([#2810](https://github.com/GenericMappingTools/pygmt/pull/2810))
> * Bump the GMT version in CI to 6.5.0 ([#2962](https://github.com/GenericMappingTools/pygmt/pull/2962))
> * Bump ghostscript to 10.02.1 ([#2694](https://github.com/GenericMappingTools/pygmt/pull/2694))
> * Benchmark grdsample, grdfilter and sph2grd with fixed cores ([#2945](https://github.com/GenericMappingTools/pygmt/pull/2945))
> * Add the validate_output_table_type function to check the output_type parameter ([#2772](https://github.com/GenericMappingTools/pygmt/pull/2772))
> * Add the full changelog link to the release drafter template ([#2838](https://github.com/GenericMappingTools/pygmt/pull/2838))
> * Add support for Python 3.12 ([#2711](https://github.com/GenericMappingTools/pygmt/pull/2711))
> * Add Zenodo's GMT community to the maintainer's onboarding list ([#2761](https://github.com/GenericMappingTools/pygmt/pull/2761))
> * Add Mypy for static type checking ([#2808](https://github.com/GenericMappingTools/pygmt/pull/2808))

Reviewer 2: Add "Switch away from Stamen basemaps" and "Figure.shift_origin: Improve docstrings, add inline examples and add type hints" (re-formulated for consistency), and do some sorting. Feel free to make further adjustments to the sorting.

```suggestion
### Maintenance

* pyarrow: Check compatibility of pyarrow-backed pandas objects with numeric dtypes ([#2774](https://github.com/GenericMappingTools/pygmt/pull/2774))
* Switch away from Stamen basemaps ([#2717](https://github.com/GenericMappingTools/pygmt/pull/2717))
* Add the "validate_output_table_type" function to check the "output_type" parameter ([#2772](https://github.com/GenericMappingTools/pygmt/pull/2772))
* Create "skip_if_no" helper function to skip tests when missing a package ([#2883](https://github.com/GenericMappingTools/pygmt/pull/2883))
* Fix "fixture_xr_image" to open "earth_day_01d_p" directly with rioxarray ([#2963](https://github.com/GenericMappingTools/pygmt/pull/2963))
* Improve the way to import optional modules ([#2809](https://github.com/GenericMappingTools/pygmt/pull/2809))
* Move variable `__gmt_version__` to pygmt.clib to avoid cyclic-import errors ([#2713](https://github.com/GenericMappingTools/pygmt/pull/2713))
* Refactor the internal "_load_remote_dataset function" to simplify datasets' definitions ([#2917](https://github.com/GenericMappingTools/pygmt/pull/2917))
* Benchmark grdsample, grdfilter and sph2grd with fixed cores ([#2945](https://github.com/GenericMappingTools/pygmt/pull/2945))
* Setup Continuous Benchmarking workflow with pytest-codspeed ([#2908](https://github.com/GenericMappingTools/pygmt/pull/2908))
* Update output shape and mean values from some x2sys_cross tests ([#2986](https://github.com/GenericMappingTools/pygmt/pull/2986))
* Bump the GMT version in CI to 6.5.0 ([#2962](https://github.com/GenericMappingTools/pygmt/pull/2962))
* Bump ghostscript to 10.02.1 ([#2694](https://github.com/GenericMappingTools/pygmt/pull/2694))
* Add support for Python 3.12 ([#2711](https://github.com/GenericMappingTools/pygmt/pull/2711))
* NEP29: Test PyGMT on NumPy 1.26 ([#2692](https://github.com/GenericMappingTools/pygmt/pull/2692))
* CI: Trigger the cache_data workflow in PRs if cache files are added/deleted/updated ([#2939](https://github.com/GenericMappingTools/pygmt/pull/2939))
* CI: Test GMT dev version on Windows by building from source ([#2773](https://github.com/GenericMappingTools/pygmt/pull/2773))
* CI: Set cache-downloads to false to speedup the "Setup Micromamba" step ([#2946](https://github.com/GenericMappingTools/pygmt/pull/2946))
* CI: Run certain GitHub Actions workflows on official repo only ([#2951](https://github.com/GenericMappingTools/pygmt/pull/2951))
* CI: Run benchmarks if PR is labeled with "run/benchmark" ([#2958](https://github.com/GenericMappingTools/pygmt/pull/2958))
* CI: Run "GMT Dev Tests" if PR is labeled with "run/test-gmt-dev" ([#2960](https://github.com/GenericMappingTools/pygmt/pull/2960))
* CI: Require at least one code block separator for example files in the Style Checks workflow ([#2810](https://github.com/GenericMappingTools/pygmt/pull/2810))
* Add Mypy for static type checking ([#2808](https://github.com/GenericMappingTools/pygmt/pull/2808))
* TYP: Improve the doc style for type hints ([#2813](https://github.com/GenericMappingTools/pygmt/pull/2813))
* TYP: Add type hints support for pygmt.datasets.load_sample_data ([#2859](https://github.com/GenericMappingTools/pygmt/pull/2859))
* TYP: Add type hints for the "registration" parameter in pygmt.datasets.load_* functions ([#2867](https://github.com/GenericMappingTools/pygmt/pull/2867))
* TYP: Add type hints for "data_source" in load_earth_relief and load_earth_magnetic_anomaly functions ([#2849](https://github.com/GenericMappingTools/pygmt/pull/2849))
* TYP: Add type hints for parameters of Figure.timestamp([#2890](https://github.com/GenericMappingTools/pygmt/pull/2890))
* TYP: Add type hints for "terminator" in Figure.solar and simplify codes ([#2881](https://github.com/GenericMappingTools/pygmt/pull/2881))
* TYP: Add type hints for parameters of Figure.shift_origin, improve docstrings, and add inline examples ([#2879](https://github.com/GenericMappingTools/pygmt/pull/2879))
* doc: Convert the installation guides source code from ReST to Markdown ([#2992](https://github.com/GenericMappingTools/pygmt/pull/2992))
* doc: Convert overview source code from ReST to Markdown ([#2953](https://github.com/GenericMappingTools/pygmt/pull/2953))
* doc: Move compatibility table from README to separate file ([#2862](https://github.com/GenericMappingTools/pygmt/pull/2862))
* Enable the PDF format documentation in the ReadTheDocs site ([#2876](https://github.com/GenericMappingTools/pygmt/pull/2876))
* Add the full changelog link to the release drafter template ([#2838](https://github.com/GenericMappingTools/pygmt/pull/2838))
* Release Drafter: Automatically replace GitHub handles with names and links ([#2777](https://github.com/GenericMappingTools/pygmt/pull/2777))
* Exclude CODE_OF_CONDUCT.md, AUTHORSHIP.md and pygmt/tests directory from distributions ([#2957](https://github.com/GenericMappingTools/pygmt/pull/2957))
* Add Zenodo's GMT community to the maintainer's onboarding list ([#2761](https://github.com/GenericMappingTools/pygmt/pull/2761))
* Use ruff to lint and format codes, and remove flakeheaven/isort/black/blackdoc ([#2741](https://github.com/GenericMappingTools/pygmt/issues/2741))
* Use codespell to check common misspellings ([#2673](https://github.com/GenericMappingTools/pygmt/pull/2673))
* Use "# %%" as code block separators in examples ([#2662](https://github.com/GenericMappingTools/pygmt/pull/2662))
```

# Changelog

## Release v0.11.0 (2024/02/01)

[![Digital Object Identifier for PyGMT v0.11.0](https://zenodo.org/badge/DOI/10.5281/zenodo.10578540.svg)](https://doi.org/10.5281/zenodo.10578540)

### Highlights

* Intro for table inputs in PyGMT

> * Gallery example for Chorophleth map

Reviewer 3: ```suggestion
* Gallery example for Choropleth map
```

> * Easy access to remote datasets of planetary reliefs (Moon, Venus, Mars, Mercury, Pluto)

Author: Implemented in #3028 and #3027

> * Easy access to remote datasets of planetary reliefs (Moon, Venus, Mars, Mercury, Pluto)

Reviewer 2: Great 🙂! Thanks (including updating the release announcement)!

> * Intro for table inputs in PyGMT and gallery example for choropleth map
> * Easy access to remote datasets of planetary reliefs (Moon, Venus, Mars, Mercury, Pluto)
> * Faster PyGMT by ~0.1 seconds for each module call

Reviewer: Link to the PRs?
```suggestion
* Tutorial for table inputs ([#2722](https://github.com/GenericMappingTools/pygmt/pull/2722)) and gallery example for choropleth map ([#2796](https://github.com/GenericMappingTools/pygmt/pull/2796))
* Easy access to remote datasets of planetary reliefs (Moon, Venus, Mars, Mercury, Pluto) ([#2674](https://github.com/GenericMappingTools/pygmt/pull/2674), [#2906](https://github.com/GenericMappingTools/pygmt/pull/2906), [#2847](https://github.com/GenericMappingTools/pygmt/pull/2847))
* Faster PyGMT by ~0.1 seconds for each module call ([#2930](https://github.com/GenericMappingTools/pygmt/pull/2930))
```

> * Intro for table inputs in PyGMT
> * Gallery example for Chorophleth map
> * Easy access to remote datasets of planetary reliefs (Moon, Venus, Mars, Mercury, Pluto)
> * `Figure.text`: Allow lists as input type for the `angle`, `font`, and `justify` parameters

Author: I agree with @yvonnefroehlich that 2-4 highlights are a good number and have applied the suggestions by @yvonnefroehlich and @weiji14.

I also would like to highlight the changes in PR https://github.com/GenericMappingTools/pygmt/pull/2930, which speed up each module call by about 0.1 seconds. The performance improvement seems small but it can significantly speed up PyGMT scripts that are not computationally intensive. For example, changes in PR #2930 have reduced our CI testing times from 250 seconds to 85 seconds (https://github.com/GenericMappingTools/pygmt/pull/2930#issue-2058474856).

If this change is highlighted, then maybe remove the highlights about Chorophleth and `Figure.text`?

```suggestion
* Intro for table inputs in PyGMT
* Easy access to remote datasets of planetary reliefs (Moon, Venus, Mars, Mercury, Pluto)
* Faster PyGMT by ~0.1 seconds for each module call
```

Reviewer: We can combine the tutorial (table inputs) and gallery example (choropleth) in one line perhaps, and remove the `Figure.text` one

Reviewer 2: I suggested the `Figure.text` improvement, because I was hoping to complete PR #2760. But as this will be unfortunately not the case, I feel we can remove this from the highlights section and move it to the nice enhancements section in the release announcement.

Reviewer 2: Regarding the choropleth map: So far, I know, it is not so straight forward to create such a visualization directly in GMT. Also, we show here the integration with geopandas. Thus, I feel it would be nice to mention it.

### New Features

* Support timedelta64 dtype as input ([#2884](https://github.com/GenericMappingTools/pygmt/pull/2884))
* Figure.text: Support passing in a list of angle/font/justify values ([#2720](https://github.com/GenericMappingTools/pygmt/pull/2720))
* Figure.savefig: Support saving figures in PPM (.ppm) format ([#2771](https://github.com/GenericMappingTools/pygmt/pull/2771))
* Figure.savefig: Support generating GeoTIFF file (with extension '.tiff') ([#2698](https://github.com/GenericMappingTools/pygmt/pull/2698))
* Figure.savefig: Add the 'worldfile' parameter to write a companion world file for raster images ([#2766](https://github.com/GenericMappingTools/pygmt/pull/2766))

> * Add load_venus_relief to load Venus relief dataset in various resolutions and registrations ([#2906](https://github.com/GenericMappingTools/pygmt/pull/2906))
> * Add load_moon_relief to load Moon relief dataset in various resolutions and registrations ([#2674](https://github.com/GenericMappingTools/pygmt/pull/2674))
> * Add load_mars_relief to load Mars relief dataset in various resolutions and registrations ([#2847](https://github.com/GenericMappingTools/pygmt/pull/2847))

Reviewer: Remove since added to highlights
```suggestion
```

### Enhancements

* pyarrow: Support date32[day] and date64[ms] dtypes in pandas objects ([#2845](https://github.com/GenericMappingTools/pygmt/pull/2845))
* geopandas: Correctly handle columns with integer values bigger than the largest 32-bit integer ([#2841](https://github.com/GenericMappingTools/pygmt/pull/2841))
* Set GMT_SESSION_NAME to a unique name on Windows for multiprocessing support ([#2938](https://github.com/GenericMappingTools/pygmt/pull/2938))
* Improve the error messages for unsupported numpy dtypes ([#2856](https://github.com/GenericMappingTools/pygmt/pull/2856))

> * Improve performance by avoiding loading the GMT library repeatedly ([#2930](https://github.com/GenericMappingTools/pygmt/pull/2930))

Reviewer: ```suggestion
```

* Figure.meca: Fix typo pricipal_axis -> principal_axis ([#2940](https://github.com/GenericMappingTools/pygmt/pull/2940))

### Documentation

* Use consistent description for the 'outgrid' parameter ([#2874](https://github.com/GenericMappingTools/pygmt/pull/2874))
* Switch away from Stamen basemaps ([#2717](https://github.com/GenericMappingTools/pygmt/pull/2717))
* Recommend Miniforge instead of Mambaforge ([#2833](https://github.com/GenericMappingTools/pygmt/pull/2833))
* Improve the onboarding access checklist for contributors, maintainers and administrators ([#2656](https://github.com/GenericMappingTools/pygmt/pull/2656))
* Gallery example "Legend": Update regarding input data and multi-column legends ([#2762](https://github.com/GenericMappingTools/pygmt/pull/2762))
* Figure.shift_origin: Improve docstrings, add inline examples and add type hints ([#2879](https://github.com/GenericMappingTools/pygmt/pull/2879))
* Figure.plot: Update docstring for "connection" parameter to GMT 6.5 ([#2994](https://github.com/GenericMappingTools/pygmt/pull/2994))
* Add the Japanese "PyGMT-HOWTO" tutorial to "External Resources" ([#2743](https://github.com/GenericMappingTools/pygmt/pull/2743))

> * Add gallery example showing usage of polygon objects from a geopandas.GeoDataFrame (choropleth map) ([#2796](https://github.com/GenericMappingTools/pygmt/pull/2796))

Reviewer: Remove since added to highlights
```suggestion
```

* Add gallery example showing how to adjust line segment ends (caps and joints) ([#3015](https://github.com/GenericMappingTools/pygmt/pull/3015))
* Add gallery example for plotting connection lines ("connection" parameter of Figure.plot) ([#2999](https://github.com/GenericMappingTools/pygmt/pull/2999))
* Add gallery example "Scale bar" ([#2822](https://github.com/GenericMappingTools/pygmt/pull/2822))

> * Add a tutorial for working with table inputs in PyGMT ([#2722](https://github.com/GenericMappingTools/pygmt/pull/2722))

Reviewer: Remove since added to highlights
```suggestion
```

# Changelog

## Release v0.11.0 (20YY/MM/DD)

> [![Digital Object Identifier for PyGMT v0.11.0](https://zenodo.org/badge/DOI/10.5281/zenodo.3781524.svg)](https://doi.org/10.5281/zenodo.3781524)

Reviewer: ```suggestion
[![Digital Object Identifier for PyGMT v0.11.0](https://zenodo.org/badge/DOI/10.5281/zenodo.10578540.svg)](https://doi.org/10.5281/zenodo.10578540)
```

### Highlights

> *

Reviewer 2: Here are some ideas for the Highlights section:
```suggestion
* Intro for table inputs in PyGMT
* Gallery example for Chorophleth map
* Easy access to remote datasets of planetary reliefs (Moon, Venus, Mars, Mercury, Pluto)
* `Figure.savefig`: Support generating GeoTIFF files
* `Figure.text`: Allow lists as input type for the `angle`, `font`, and `justify` parameters
* Start adding type hints support
```
Feel free to select the preferred items (I feel 2-4 highlights are a good number) and to improve the formulations.

Reviewer: Maybe omit the fig.savefig one, and leave out the type hints too (there's not many of them yet).

* Add common alias cores (x) for grdlandmask ([#2944](https://github.com/GenericMappingTools/pygmt/pull/2944))

### Deprecations

* Rename API function Session.open_virtual_file to Session.open_virtualfile (remove in v0.15.0) ([#2996](https://github.com/GenericMappingTools/pygmt/pull/2996))
* NEP29: Set minimum required version to NumPy 1.23+ ([#2991](https://github.com/GenericMappingTools/pygmt/pull/2991))
* Figure.timestamp: Deprecate parameter 'justification' to 'justify' (remove in v0.13.0) ([#3002](https://github.com/GenericMappingTools/pygmt/pull/3002))
* Figure.grdimage: Remove the unsupported 'img_out'/'A' parameter ([#2907](https://github.com/GenericMappingTools/pygmt/pull/2907))

### Bug Fixes

* pygmt.which: Fix the bug when passing multiple files ([#2726](https://github.com/GenericMappingTools/pygmt/pull/2726))
* pygmt.filter1d: Fix the bug that the first line is read as headers ([#2780](https://github.com/GenericMappingTools/pygmt/pull/2780))
* clib: Fix the bug when passing multiple columns of strings with variable lengths to the GMT C API ([#2719](https://github.com/GenericMappingTools/pygmt/pull/2719))

> * Let kwargs_to_strings work with default values and postional arguments ([#2826](https://github.com/GenericMappingTools/pygmt/pull/2826))

Reviewer 2: ```suggestion
* Let kwargs_to_strings work with default values and positional arguments ([#2826](https://github.com/GenericMappingTools/pygmt/pull/2826))
```

* Figure.plot: Update docstring for "connection" parameter to GMT 6.5 ([#2994](https://github.com/GenericMappingTools/pygmt/pull/2994))
* Examples: Replace deprecated "np.random.seed()" with "np.random.Generator()" (NPY002 warning by ruff) ([#2781](https://github.com/GenericMappingTools/pygmt/pull/2781))
* Add the Japanese "PyGMT-HOWTO" tutorial to "External Resources" ([#2743](https://github.com/GenericMappingTools/pygmt/pull/2743))
* Add gallery example showing usage of polygon objects from a geopandas.GeoDataFrame (choropleth map) ([#2796](https://github.com/GenericMappingTools/pygmt/pull/2796))
* Add gallery example showing how to adjust line segment ends (caps and joints) ([#3015](https://github.com/GenericMappingTools/pygmt/pull/3015))
* Add gallery example for plotting connection lines ("connection" parameter of Figure.plot) ([#2999](https://github.com/GenericMappingTools/pygmt/pull/2999))
* Add gallery example "Scale bar" ([#2822](https://github.com/GenericMappingTools/pygmt/pull/2822))
* Add a tutorial for working with table inputs in PyGMT ([#2722](https://github.com/GenericMappingTools/pygmt/pull/2722))

> * Add a 'Common installation issues' section to the install guides ([#2868](https://github.com/GenericMappingTools/pygmt/pull/2868))

Author: ```suggestion
```

### Maintenance

* pyarrow: Check compatibility of pyarrow-backed pandas objects with numeric dtypes ([#2774](https://github.com/GenericMappingTools/pygmt/pull/2774))
* doc: Convert the installation guides source code from ReST to Markdown ([#2992](https://github.com/GenericMappingTools/pygmt/pull/2992))
* doc: Convert overview source code from ReST to Markdown ([#2953](https://github.com/GenericMappingTools/pygmt/pull/2953))
* Use ruff to lint and format codes, and remove flakeheaven/isort/black/blackdoc ([#2741](https://github.com/GenericMappingTools/pygmt/issues/2741))
* Use codespell to check common misspellings ([#2673](https://github.com/GenericMappingTools/pygmt/pull/2673))

> * Use GITHUB_TOKEN instead of PAT in "Slash Command Dispatch" and "GMT Dev Tests" workflows ([#2950](https://github.com/GenericMappingTools/pygmt/pull/2950))

Author: ```suggestion
```

* Update output shape and mean values from some x2sys_cross tests ([#2986](https://github.com/GenericMappingTools/pygmt/pull/2986))
* TYP: Improve the doc style for type hints ([#2813](https://github.com/GenericMappingTools/pygmt/pull/2813))
* TYP: Add type hints support for pygmt.datasets.load_sample_data ([#2859](https://github.com/GenericMappingTools/pygmt/pull/2859))
* TYP: Add type hints for the 'registration' parameter in pygmt.datasets.load_* functions ([#2867](https://github.com/GenericMappingTools/pygmt/pull/2867))
* TYP: Add type hints for 'data_source' in load_earth_relief and load_earth_magnetic_anomaly functions ([#2849](https://github.com/GenericMappingTools/pygmt/pull/2849))
* Setup Continuous Benchmarking workflow with pytest-codspeed ([#2908](https://github.com/GenericMappingTools/pygmt/pull/2908))
* Release Drafter: Automatically replace GitHub handles with names and links ([#2777](https://github.com/GenericMappingTools/pygmt/pull/2777))
* Refactor the internal _load_remote_dataset function to simplify datasets' definitions ([#2917](https://github.com/GenericMappingTools/pygmt/pull/2917))

> * RUF010: Avoid calling str() within an f-string ([#2871](https://github.com/GenericMappingTools/pygmt/pull/2871))
> * RUF005: Use the unpacking operator to concatenate collections ([#2869](https://github.com/GenericMappingTools/pygmt/pull/2869))

Author: ```suggestion
```

* NEP29: Test PyGMT on NumPy 1.26 ([#2692](https://github.com/GenericMappingTools/pygmt/pull/2692))

> * Move variable __gmt_version__ to pygmt.clib to avoid cyclic-import errors ([#2713](https://github.com/GenericMappingTools/pygmt/pull/2713))

Reviewer: ```suggestion
* Move variable `__gmt_version__` to pygmt.clib to avoid cyclic-import errors ([#2713](https://github.com/GenericMappingTools/pygmt/pull/2713))
```

* Create skip_if_no helper function to skip tests when missing a package ([#2883](https://github.com/GenericMappingTools/pygmt/pull/2883))
* CI: Trigger the cache_data workflow in PRs if cache files are added/deleted/updated ([#2939](https://github.com/GenericMappingTools/pygmt/pull/2939))
* CI: Test GMT dev version on Windows by building from source ([#2773](https://github.com/GenericMappingTools/pygmt/pull/2773))
* CI: Set cache-downloads to false to speedup the "Setup Micromamba" step ([#2946](https://github.com/GenericMappingTools/pygmt/pull/2946))
* CI: Run certain GitHub Actions workflows on official repo only ([#2951](https://github.com/GenericMappingTools/pygmt/pull/2951))
* CI: Run benchmarks if PR is labeled with "run/benchmark" ([#2958](https://github.com/GenericMappingTools/pygmt/pull/2958))
* CI: Run "GMT Dev Tests" if PR is labeled with 'run/test-gmt-dev' ([#2960](https://github.com/GenericMappingTools/pygmt/pull/2960))
* CI: Require at least one code block separator for example files in the Style Checks workflow ([#2810](https://github.com/GenericMappingTools/pygmt/pull/2810))

> * CI: Cache the earth_day_01d_p grid ([#2937](https://github.com/GenericMappingTools/pygmt/pull/2937))

Author: ```suggestion
```

* Bump the GMT version in CI to 6.5.0 ([#2962](https://github.com/GenericMappingTools/pygmt/pull/2962))
* Bump ghostscript to 10.02.1 ([#2694](https://github.com/GenericMappingTools/pygmt/pull/2694))
* Benchmark grdsample, grdfilter and sph2grd with fixed cores ([#2945](https://github.com/GenericMappingTools/pygmt/pull/2945))
* Add the validate_output_table_type function to check the output_type parameter ([#2772](https://github.com/GenericMappingTools/pygmt/pull/2772))
* Add the full changelog link to the release drafter template ([#2838](https://github.com/GenericMappingTools/pygmt/pull/2838))
* Add support for Python 3.12 ([#2711](https://github.com/GenericMappingTools/pygmt/pull/2711))
* Add Zenodo's GMT community to the maintainer's onboarding list ([#2761](https://github.com/GenericMappingTools/pygmt/pull/2761))
* Add Mypy for static type checking ([#2808](https://github.com/GenericMappingTools/pygmt/pull/2808))

> **Full Changelog**: https://github.com/GenericMappingTools/pygmt/compare/v0.10.0...v0.11.0

Reviewer: This is new! Do we need to add this? If so, we need to make the link clickable
```suggestion
**Full Changelog**: [https://github.com/GenericMappingTools/pygmt/compare/v0.10.0...v0.11.0](https://github.com/GenericMappingTools/pygmt/compare/v0.10.0...v0.11.0)
```
